src/_pytest/python.py

```diff
@@ -1059,13 +1058,11 @@ def _idval(val, argname, idx, idfn, item, config):
             s = idfn(val)
         except Exception as e:
             # See issue https://github.com/pytest-dev/pytest/issues/2169
-            msg = (
-                "While trying to determine id of parameter {} at position "
-                "{} the following exception was raised:\n".format(argname, idx)
-            )
+            msg = "{}: error raised while trying to determine id of parameter '{}' at position {}\n"
+            msg = msg.format(item.nodeid, argname, idx)
+            # we only append the exception type and message because on Python 2 reraise does nothing
             msg += "  {}: {}\n".format(type(e).__name__, e)
-            msg += "This warning will be an error error in pytest-4.0."
-            item.warn(RemovedInPytest4Warning(msg))
+            six.reraise(type(e), ValueError(msg))
```

Comment:
should use https://pythonhosted.org/six/#six.raise_from

Reply:
True, fixed.